pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20955868?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20955868?buildTab=artifacts#/) on master @ [ccb2b5ddab424fe5a097c2ddff8e3811881d9a59](https://github.com/cockroachdb/cockroach/commits/ccb2b5ddab424fe5a097c2ddff8e3811881d9a59):


```
     "public.table_w0_67",
     "table_w0_67_pkey"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w0_67 SET (sql_stats_histogram_samples_count=INVALID_VAL)"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: invalid integer value for sql_stats_histogram_samples_count: could not parse "invalid_val" as type int: strconv.ParseInt: parsing "invalid_val": invalid syntax (SQLSTATE 22P02)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3399
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: invalid integer value for sql_stats_histogram_samples_count: could not parse "invalid_val" as type int: strconv.ParseInt: parsing "invalid_val": invalid syntax (SQLSTATE 22P02)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 20
Total Schema Statements Succeeded = 20
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 10
Total Transactions Rolled Back = 16
Total Transactions Executed = 26
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload764710980
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload764710980
--- FAIL: TestWorkload (32.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #160598 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.4.3-rc release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58527

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -5801,7 +5801,11 @@ func (og *operationGenerator) setTableStorageParam(
 	}
 
 	if !isValid {
+		// Depending on the type of the parameter, different error codes are
+		// returned, so we add all possible ones.
 		stmt.expectedExecErrors.add(pgcode.InvalidParameterValue)
+		stmt.expectedExecErrors.add(pgcode.InvalidTextRepresentation)
+		stmt.expectedExecErrors.add(pgcode.InvalidDatetimeFormat)
 	} else if isSchemaLockedParam(param) {
 		// set schema_locked cannot be combined with other operations
 		// and will return an error if it does.
@@ -5859,7 +5863,7 @@ func (og *operationGenerator) resetTableStorageParam(
 
 // randBoolString returns a random boolean value as a string
 func (og *operationGenerator) randBoolString() string {
-	boolVals := []string{"true", "false", "'on'", "'off'", "1", "0"}
+	boolVals := []string{"true", "false", "'on'", "'off'", "'1'", "'0'"}
 	return boolVals[og.randIntn(len(boolVals))]
 }
 
@@ -5929,7 +5933,12 @@ func (og *operationGenerator) findTimestampColumn(
 }
 
 // randStorageParam returns a random storage parameter (excluding the ttl base params)
-func (og *operationGenerator) randStorageParam() (string, string, bool, error) {
+func (og *operationGenerator) randStorageParam() (
+	param string,
+	value string,
+	isValid bool,
+	_ error,
+) {
 	params := []struct {
 		name     string
 		valueGen func() string

